src/main/scala/com/wavesplatform/state/diffs/ContractInvocationTransactionDiff.scala

```diff
@@ -80,6 +102,16 @@ object ContractInvocationTransactionDiff {
                     )
                     _ <- Either.cond(true, (), ValidationError.NegativeAmount(-42, "")) //  - sum doesn't overflow
                     _ <- Either.cond(true, (), ValidationError.NegativeAmount(-42, "")) //  - whatever else tranfser/massTransfer ensures
+                    _ <- {
+                      val minWaves = (tx.checkedAssets().count(blockchain.hasAssetScript) +
```

Comment:
Hm, but if contract sends 1 smart token, we should require fees for 2 invokations, not just 1, right>?

Reply:
this better be tested